modules/friendlytag.js

```diff
@@ -1528,13 +1528,13 @@ Twinkle.tag.callbacks = {
 					// capture template(s)
 					'(?:((?:\\s*' +
 					// AfD is special, as the tag includes html comments before and after the actual template
-					'(?:<!--.*AfD.*\\n\\{\\{Article for deletion\\/dated.*\\}\\}\\n<!--.*\\n<!--.*AfD.*(?:\\s*\\n))?|' + // trailing whitespace/newline needed since this subst's a newline
+					'(?:<!--.*AfD.*\\n\\{\\{(?:Article for deletion\\/dated|AfDM).*\\}\\}\\n<!--.*(?:\\n<!--.*)?AfD.*(?:\\s*\\n))?|' + // trailing whitespace/newline needed since this subst's a newline
 					// begin template format
 					'\\{\\{\\s*(?:' +
 					// CSD
 					'db|delete|db-.*?|speedy deletion-.*?|' +
 					// PROD
-					'(?:proposed deletion|prod blp)\\/dated\\n(?:\\s+\\|(?:concern|user|timestamp|help).*)+|' +
+					'(?:proposed deletion|prod blp)\\/dated(?:\\s+\\|(?:concern|user|timestamp|help).*)+|' +
```

Comment:
Not sure which was intentional, but this didn't line up with your commit comment.

```suggestion
					'(?:proposed deletion|prod blp)\\/dated(?:\\s*\\|(?:concern|user|timestamp|help).*)+|' +
```

At any rate, what's the impetus behind making this change?  The templates should always be substituted, so would this change just cover the case of someone/something manually adding `{{proposed deletion/dated|concern=concern|timestamp={{subst:CURRENTTIMESTAMP}}}}` rather than `{{subst:prod|concern=concern}}`?

Reply:
`\\s*` is more correct, but of course as long as templates are substed, it doesn't matter. The impetus behind the change is that it makes the regex (slightly) simpler while also accounting for the case of someone manually adding {{proposed deletion/dated}}.

Reply:
Of course, here in tag module, all this hardly matters since the world won't freeze over if a tag gets added above a prod notice rather than below. But complicated regexes like this are likely to be reused/adapted by people in other more important applications (like I am doing in my rfud-helper script, where it's necessary that all deletion tags be removed from a page after it's been restored). So it's good to get the regex perfectly right.

Reply:
Okay, that's fine.  Generally I think it's good to be as parsimonious as reasonable.